wsclient/src/main/java/org/hpccsystems/ws/client/extended/axis2/HPCCWsAttributesClient.java

```diff
@@ -0,0 +1,847 @@
+     *
+     */
+    private void initWsAttributesClientStub(Connection connection)
+    {
+        try
+        {
+            stub = setStubOptions(new WsAttributesStub(connection.getBaseUrl()+WSATTRIBUTESWSDLURI), connection);
+        }
+        catch (AxisFault e)
+        {
+            log.error("Could not initialize WsAttributesStub - Review all HPCC connection values");
+            e.printStackTrace();
+        }
+        catch (Exception e)
+        {
+            log.error("Could not initialize WsAttributesStub - Review all HPCC connection values");
+            if (!e.getLocalizedMessage().isEmpty())
+            {
+                initErrMessage = e.getLocalizedMessage();
+                log.error(e.getLocalizedMessage());
+            }
+        }
+    }
+
+    /**
+     * Check whether an attribute exists
+     *
+     * @param modulename
+     *            - ECL Module name containing the attribute to check
+     * @param attributename
+     *            - Attribute to check
+     * @return true if the attribute exists, false if it does not
+     * @throws Exception
+     */
+    public boolean attributeExists(String modulename, String attributename, String type) throws Exception
+    {
+        List<ECLAttributeWrapper>items = findItems(modulename, attributename, type, null, null, null);
+        if (items.size()>0)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Search for and retrieve the attribute metadata for attributes from a repository
+     *
+     * @param modulename
+     *            - modulename of attributes to find. Entire name will be matched.
+     * @param attributename
+     *            - attributename of attributes to find. Entire name will be matched.
+     * @param username
+     *            - owner of attributes to find. Entire name will be matched
+     * @param anytext
+     *            - text string to find. It is implicitly surrounded by wildcards. will match any module name containing
+     *            the pattern, attribute name containing the pattern, or attribute containing ecl that contains the
+     *            pattern
+     * @param changedSince
+     *            - only return attributes changes since this time
+     * @return a list of ECLAttributeInfo
+     * @throws Exception
+     */
+    public List<ECLAttributeWrapper> findItems(String modulename, String attributename, String type, String username, String anytext, String changedSince) throws Exception
+    {
+        verifyStub(); //Throws if stub not available
+
+        if (modulename == null && attributename == null && username == null &&
+                type==null && anytext == null && changedSince == null)
+        {
+            throw new Exception("At least one find criteria is required.");
+        }
+
+        FindAttributes params = new FindAttributes();
+
+        if (modulename != null)
```

Comment:
We throw an exception above if these params are null. Should we remove the exception or remove the null checks?

Reply:
It looks like we throw the exception if all params are null, and then we avoid setting the individual param if it is null